src/ai/backend/gateway/session.py

```diff
@@ -700,6 +701,27 @@ def interrupt_wait(ctx: Any, event_name: str, agent_id: AgentId,
     await app['registry'].handle_stats(agent_id, kern_stats)
 
 
+async def handle_kernel_log(app: web.Application, agent_id: AgentId, event_name: str,
+                            raw_kernel_id: str, container_id: str):
+    dbpool = app['dbpool']
+    connection: aioredis.Redis = await redis.connect_with_retries(
+                app['config']['redis']['addr'].as_sockaddr(),
+                db=REDIS_STREAM_DB,
+                password=(app['config']['redis']['password']
+                          if app['config']['redis']['password'] else None),
+                encoding=None,
+    )
+    logs = await redis.execute_with_retries(
+            lambda: connection.lrange(f'containerlog.{container_id}', 0, -1))
```

Comment:
Please split this into multiple calls of LPOP.
The reason to use Redis lists is *not* to block the connection while we are sending/receiving large data to/from Redis.